src/dodal/common/beamlines/device_factory.py

```diff
@@ -0,0 +1,112 @@
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Generic, ParamSpec, TypeVar
+
+from bluesky.run_engine import call_in_bluesky_event_loop
+from ophyd_async.core import DEFAULT_TIMEOUT, Device
+
+from dodal.common.beamlines.beamline_utils import ACTIVE_DEVICES
+
+P = ParamSpec("P")
+T = TypeVar("T", bound=Device)
+_skip = bool | Callable[[], bool]
+
+
+@dataclass
+class DeviceInitializationConfig:
+    """
+    default_eager: connect or raise Exception at startup
+    default_set_name: use factory name as name of device
+    default_timeout: timeout for connecting to the device
+    default_mock: use Signals with mock backends for device
+    default_skip: mark the factory to be (conditionally) skipped
```

Comment:
Please can these docstrings be updated to match the names of the attributes.

Also, what does it mean to `skip` a device?

Reply:
`skip` is for when all devices on a beamline are being created automaticlaly, e.g. by blueapi. We parse the file for all device factory functions, then call those that do not have skip evaluate to True. Therefore, skip isn't exposed as one of the function params for the decorated function, as if you would want the device to not be created you wouldn't call the function. It's only exposed as a field on the `DeviceInitializationController` that blueapi etc. can check.